controllers/machine_controller.go

```diff
@@ -275,13 +294,26 @@ func (r *MachineReconciler) reconcileDelete(ctx context.Context, cluster *cluste
 		if annotations.HasWithPrefix(clusterv1.PreDrainDeleteHookAnnotationPrefix, m.ObjectMeta.Annotations) {
 			return ctrl.Result{}, nil
 		}
```

Comment:
Should we have a condition notifying that deletion is blocked on pre-drain hook?

Reply:
Same goes for `PreTerminateDeleteHookAnnotationPrefix` on line 291 in machine_controller.go

Reply:
done!